docs/bigquery/snippets.py

```diff
@@ -2579,6 +2579,178 @@ def test_query_no_cache(client):
     # [END bigquery_query_no_cache]
 
 
+def test_query_external_gcs_temporary_table(client):
+    # [START bigquery_query_external_gcs_temp]
+    # from google.cloud import bigquery
+    # client = bigquery.Client()
+
+    # Configure the external data source and query job
+    external_config = bigquery.ExternalConfig('CSV')
+    external_config.source_uris = [
+        'gs://cloud-samples-data/bigquery/us-states/us-states.csv',
+    ]
+    external_config.schema = [
+        bigquery.SchemaField('name', 'STRING'),
+        bigquery.SchemaField('post_abbr', 'STRING')
+    ]
+    external_config.options.skip_leading_rows = 1  # optionally skip header row
+    table_id = 'us_states'
+    job_config = bigquery.QueryJobConfig()
+    job_config.table_definitions = {table_id: external_config}
+
+    # Example query to find states starting with 'W'
+    sql = 'SELECT * FROM {} WHERE name like "W%"'.format(table_id)
+
+    query_job = client.query(sql, job_config=job_config)  # API request
+
+    w_states = list(query_job)  # Waits for query to finish
+    print('There are {} states with names starting with W.'.format(
+        len(w_states)))
+    # [END bigquery_query_external_gcs_temp]
+    assert len(w_states) == 4
+
+
+def test_query_external_gcs_permanent_table(client, to_delete):
+    dataset_id = 'query_external_gcs_{}'.format(_millis())
+    dataset = bigquery.Dataset(client.dataset(dataset_id))
+    client.create_dataset(dataset)
+    to_delete.append(dataset)
+
+    # [START bigquery_query_external_gcs_perm]
```

Comment:
Should creating the permanent table be a separate sample from querying one? I guess the query part is pretty small, so maybe not.